### ISSUE TITLE
process multiple public keys from WKD

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-android#1396]
> For now, our logic works just with a single key. Should I use the first matched?

All public keys returned from WKD should be processed, they should all be inserted or updated in local storage (comparing them by fingerprint), and when sending out a message, instead of choosing the best fitting public key like we did before, we should encrypt for all valid pubkeys that we have for that person. I'll file another issue.

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-android/issues/1201#issuecomment-896041422_

**Comment by DenBond7** [comment]
#1188

**Comment by tomholub** [comment]
Oh, we didn't actually do #1188 yet! Yes, that would need to be done first.

**Comment by DenBond7** [comment]
> All public keys returned from WKD should be processed, they should all be inserted or updated in local storage (comparing them by fingerprint), 

https://github.com/FlowCrypt/flowcrypt-android/blob/20cd64ba705b203b755bbdf730b4538122e19c1a/FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/viewmodel/RecipientsViewModel.kt#L332

> and when sending out a message, instead of choosing the best fitting public key like we did before, we should encrypt for all valid pubkeys that we have for that person.

https://github.com/FlowCrypt/flowcrypt-android/blob/20cd64ba705b203b755bbdf730b4538122e19c1a/FlowCrypt/src/main/java/com/flowcrypt/email/security/SecurityUtils.kt#L123

We've already added this functionality. I'm closing it.
